jitbuilder/release/src/AtomicOperations.cpp

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2016, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+
+#include <iostream>
+#include <stdlib.h>
+#include <stdint.h>
+#include <dlfcn.h>
+#include <errno.h>
+
+#include "Jit.hpp"
+#include "ilgen/TypeDictionary.hpp"
+#include "ilgen/MethodBuilder.hpp"
+#include "AtomicOperations.hpp"
+
+using std::cout;
+using std::cerr;
+
+
+int
+main(int argc, char *argv[])
+   {
+   cout << "Step 1: initialize JIT\n";
+   bool initialized = initializeJit();
+   if (!initialized)
+      {
+      cerr << "FAIL: could not initialize JIT\n";
+      exit(-1);
+      }
+
+   cout << "Step 2: define type dictionary\n";
+   TR::TypeDictionary types;
+
+   cout << "Step 3: compile two method builders\n";
+   AtomicInt32Add method(&types);
+   uint8_t *entry32 = 0;
+   int32_t rc = compileMethodBuilder(&method, &entry32);
+   if (rc != 0)
+      {
+      cerr << "First method FAIL: compilation error " << rc << "\n";
+      exit(-2);
+      }
+
+   AtomicInt64Add method2(&types);
```

Comment:
Actually, I would prefer these to be interleaved: compile 32-bit, run 32-bit, compile 64-bit, run 64-bit. I guess you could argue it either way, but I think considering 32-bit and 64-bit as two different tests is a better organization than considering it as a compilation test of 2 methods and then a runtime test of two methods.